### DISCUSSION TITLE
When deploying config, first do a diff and only do PUT when there are changes

### DISCUSSION THREAD

**Discussion by kristofre** [Dynatrace/dynatrace-configuration-as-code#227] [Archive]
At the moment, when Monaco finds a configuration that already exists, it will always do a PUT against the endpoint to update it even when there are no changes to be made.

In some cases this triggers additional workloads in Dynatrace - and additional load as well.

The suggestion is to compare - on Monaco side - the existing object on the Dynatrace side with the new object that is to be applied. If the two objects are the same then we should skip the PUT altogether.

**Comment by UnseenWizzard** [comment]
One Issue I see with this, is that what the API returns and what the API accepts on PUT/POST is not always the same. 
So a direct diff between what we GET and what we would upload is not possible, and things like metadata, ids and potentially some config specific fields would need to stripped/ignored. 

What are examples of the extra load on Dynatrace when a config is PUT?
